src/status_im/ui/screens/browser/events.cljs

```diff
@@ -10,78 +9,95 @@
             [status-im.data-store.browser :as browser-store]
             [status-im.utils.http :as http]))
 
+(re-frame/reg-fx
+ :browse
+ (fn [link]
+   (if (utils.universal-links/universal-link? link)
+     (utils.universal-links/open! link)
+     (list-selection/browse link))))
+
 (handlers/register-handler-fx
  :initialize-browsers
  [(re-frame/inject-cofx :data-store/all-browsers)]
  (fn [{:keys [db all-stored-browsers]} _]
    (let [browsers (into {} (map #(vector (:browser-id %) %) all-stored-browsers))]
      {:db (assoc db :browser/browsers browsers)})))
 
-(re-frame/reg-fx
- :browse
- (fn [link]
-   (if (utils.universal-links/universal-link? link)
-     (do
-       (utils.universal-links/open! link))
-     (list-selection/browse link))))
-
 (handlers/register-handler-fx
  :browse-link-from-message
  (fn [_ [_ link]]
    {:browse link}))
 
-(defn get-new-browser [browser now]
-  (cond-> browser
-    true
-    (assoc :timestamp now)
-    (not (:browser-id browser))
-    (assoc :browser-id (random/id))
-    (not (:name browser))
-    (assoc :name (i18n/label :t/browser))
-    (:url browser)
-    (update :url (comp js/decodeURI http/normalize-url))))
-
-(defn add-browser-fx [{:keys [db now]} browser]
-  (let [new-browser (get-new-browser browser now)]
-    {:db            (update-in db [:browser/browsers (:browser-id new-browser)]
-                               merge new-browser)
-     :data-store/tx [(browser-store/save-browser-tx new-browser)]}))
+(defn update-browser-fx [{:keys [db now]} browser]
+  (let [updated-browser (assoc browser :timestamp now)]
+    {:db            (update-in db [:browser/browsers (:browser-id updated-browser)]
+                               merge updated-browser)
+     :data-store/tx [(browser-store/save-browser-tx updated-browser)]}))
+
+(defn update-browser-history-fx [cofx browser url loading]
+  (cond
+    loading
+    nil
+
+    (get-in cofx [:db :browser/options :dont-store-history-on-nav-change?])
+    (assoc-in (update-browser-fx cofx browser)
+              [:db :browser/options :dont-store-history-on-nav-change?]
+              false)
+
+    :else
+    (let [history-index (:history-index browser)
+          history       (:history browser)
+          history-url   (try (nth history history-index) (catch js/Error _))
+          new-history   (if (not= history-url url)
+                          (conj (subvec history 0 (inc history-index))
```

Comment:
Worth a separate fn with proper name?

Reply:
too many params  and too small function don't worth it from my pov

Reply:
Small fns are never an issue :) Mostly useful to understand what this thing is doing. Or maybe a small comment?